backend.native/tests/build.gradle

```diff
@@ -402,13 +402,35 @@ Task linkTest(String name, Closure<KonanLinkTest> configureClosure) {
                 }
                 UtilsKt.dependsOnKonanBuildingTask(task, lib, target)
 
+                def dist = project.kotlinNativeDist
+                def stdlib = "$dist/klib/common/stdlib"
+                def cacheDir = project.file("${project.buildDir}/cache")
+                def stdlibCache = "$cacheDir/stdlib-cache"
+                def libPath = konanArtifacts[lib].getArtifactByTarget(targetName).absolutePath
+                def libCache = "${cacheDir.absolutePath}/${name}-cache"
+                def cacheKind = project.findProperty("test_with_cache_kind")
                 // Build an executable with library
                 program(name, targets: [targetName]) {
+                    doFirst {
```

Comment:
Is it possible that this `doFirst` is applied to the task, not to the `program`?
In this case it will be called too late.
That also might explain why the added test failes with `No cache found for library` for me.

Also a review of this commit from @PavelPunegov might be useful.

Reply:
Yes, that's true, the test is unstable now.